### PR TITLE
Shared feed data points selection

### PR DESCRIPTION
**Pull request by caiosba**
## Description

This PR introduces a new feature: shared feed data points selection. See designs and details in the referenced tickets. Technically, it means:

- Two new components added: `<FeedDataPoints />` and `<FeedDataPointsSection />`
- Unit tests added for these two new components
- Minor changes to some other feed components
- A lot of new localizable strings

References: CV2-4034 and CV2-3923.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

I implemented unit tests for the two new components I implemented, `<FeedDataPoints />` and `<FeedDataPointsSection />`.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by brianfleming** [comment]
oooooh, looking good @caiosba 
One question as I go through the review. I pulled this branch and updated my API to latest on develop and everything loads, etc. But the question is, for existing shared feeds, like the one I have locally, which was created before this work, any way for it to load the "correct" data definition, which would be just fact-checks. As you can see in the screenshot, it correctly does not let me edit the information, but it appears as though no data is being shared:

Screenshot 2024-01-11 at 2 27 44 PM

**Comment by caiosba** [comment]
Yes @brianfleming, we have a post-deployment task for this, but you can also do it in your localmente environment:

```
docker-compose exec api bundle exec rails c
Feed.where(data_points: []).update_all(data_points: [1])
```

File: src/app/components/feed/FeedContent.js [hunk 1]
```diff
@@ -1,61 +1,78 @@
...
+        /> : null }
+
+      {/* Feed is sharing both fact-checks and media/claim/requests */}
+      { dataPoints.includes(CheckFeedDataPoints.PUBLISHED_FACT_CHECKS) && dataPoints.includes(CheckFeedDataPoints.MEDIA_CLAIM_REQUESTS) ?
+        <FormattedHTMLMessage
+          id="saveFeed.feedContentBlurbFactChecksAndMedia"
+          defaultMessage="Select a custom filtered list of <strong>published fact-checks or media, claim & requests</strong> from your workspace to contribute to this shared feed. You will be able to update this list at any time."
+          description="Helper text for the feed content section when the feed is sharing fact-checks and media."
+        /> : null }
+
```
Comment (dariusk) [inline src/app/components/feed/FeedContent.js:49]: Is it possible that `dataPoints` would have none of these numbers in them? If so, should we catch that and display an error alert on `!dataPoints.includes(CheckFeedDataPoints.PUBLISHED_FACT_CHECKS) && !dataPoints.includes(CheckFeedDataPoints.MEDIA_CLAIM_REQUESTS)` ?

File: src/app/components/feed/FeedDataPoints.js [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+import PropTypes from 'prop-types';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import CheckFeedDataPoints from '../../CheckFeedDataPoints';
+import FeedDataPointsSection from './FeedDataPointsSection';
+import Alert from '../cds/alerts-and-prompts/Alert';
+import { stringHelper } from '../../customHelpers';
+import styles from './SaveFeed.module.css';
+
+const FeedDataPoints = ({ readOnly, dataPoints, onChange }) => {
+  const toggleDataPoint = (enabled, dataPoint) => {
```
Comment (dariusk) [inline src/app/components/feed/FeedDataPoints.js:11]: It took me a minute to figure out what's happening here but I like what you're doing by essentially just removing the dataPoint from the array and then adding it back in if we are toggling it on.

**Review by dariusk — APPROVED** [review]
This seems straightforward overall!

File: src/app/components/feed/FeedContent.js (hunk 1 above)
Comment (caiosba) [reply src/app/components/feed/FeedContent.js:49]: Good question! No, there is a validation to be sure that there is at least one.

**Pull request merged**